endpoints/openrtb2/auction.go

```diff
@@ -1334,6 +1339,44 @@ func validateDevice(device *openrtb2.Device) error {
 	return nil
 }
 
+func validateOrFillChannel(reqWrapper *openrtb_ext.RequestWrapper, isAmp bool) error {
+	requestExt, err := reqWrapper.GetRequestExt()
+	if err != nil {
+		return err
+	}
+	requestPrebid := requestExt.GetPrebid()
+
+	if requestPrebid == nil || requestExt.GetPrebid().Channel == nil {
+		fillChannel(reqWrapper, isAmp)
+	} else if requestPrebid.Channel.Name == "" {
+		return errors.New("ext.prebid.channel.name can't be empty")
+	}
+	return nil
+}
+
+func fillChannel(reqWrapper *openrtb_ext.RequestWrapper, isAmp bool) error {
+	requestExt, err := reqWrapper.GetRequestExt()
+	if err != nil {
+		return err
+	}
+	requestPrebid := requestExt.GetPrebid()
+
+	if requestPrebid == nil {
```

Comment:
What are your thoughts about rewriting this as:
```
var channelName string
if isAmp {
	channelName = ampChannelName
}
if reqWrapper.App != nil {
	channelName = appChannelName
}

if channelName != "" {
	if requestPrebid == nil {
		requestPrebid = &openrtb_ext.ExtRequestPrebid{}
	}
	requestPrebid.Channel = &openrtb_ext.ExtRequestPrebidChannel{Name: channelName}
	requestExt.SetPrebid(requestPrebid)
	reqWrapper.RebuildRequest()
}
return nil
```

Reply:
I like this rewrite! I'll implement it :)

Reply:
Just to confirm, we don't expect a request to come with a channel set in the AMP or APP cases? This does overwrite any channel that may be found in the original request.